lib/runtime/src/worker.rs

```diff
@@ -94,15 +95,37 @@ impl Worker {
         &self.runtime
     }
 
+    pub fn execute<F, Fut>(self, f: F) -> Result<()>
+    where
+        F: FnOnce(Runtime) -> Fut + Send + 'static,
+        Fut: Future<Output = Result<()>> + Send + 'static,
+    {
+        let runtime = self.runtime.clone();
+        runtime.secondary().block_on(self.execute_internal(f))??;
```

Comment:
I think this should be `primary`. I believe @ryanolson wanted primary to be for the main tasks (AI inference, anything on the critical path), and secondary to be for background stuff.

Reply:
Yes, that makes sense. This is implemented in the execute_internal method, where the primary runtime handles the main tasks and the secondary sets up the signal handler.

Reply:
@abrarshivani Did you get a chance to update this? No worries if you're busy, just we're very close! Would be nice to merge.

Reply:
I'm referring to this: 

https://github.com/ai-dynamo/dynamo/blob/af6576947a2f40d1fc5b515399720510d90ac24c/lib/runtime/src/worker.rs#L151

https://github.com/ai-dynamo/dynamo/blob/af6576947a2f40d1fc5b515399720510d90ac24c/lib/runtime/src/worker.rs#L153

Reply:
Got it, thank you.